pilot/pkg/config/kube/ingress/conversion_test.go

```diff
@@ -21,10 +21,123 @@ import (
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 
+	networking "istio.io/api/networking/v1alpha3"
+
 	meshconfig "istio.io/api/mesh/v1alpha1"
 	"istio.io/istio/pilot/pkg/model"
 )
 
+func TestConversion(t *testing.T) {
+	ingress := v1beta1.Ingress{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Namespace: "mock", // goes into backend full name
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "my.host.com",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Path: "/test",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "foo",
+										ServicePort: intstr.IntOrString{IntVal: 8000},
+									},
+								},
+							},
+						},
+					},
+				},
+				{
+					Host: "my2.host.com",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Path: "/test1.*",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "bar",
+										ServicePort: intstr.IntOrString{IntVal: 8000},
+									},
+								},
+							},
+						},
+					},
+				},
+				{
+					Host: "my3.host.com",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Path: "/test/*",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "bar",
+										ServicePort: intstr.IntOrString{IntVal: 8000},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	ingress2 := v1beta1.Ingress{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Namespace: "mock",
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "my.host.com",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Path: "/test2",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "foo",
+										ServicePort: intstr.IntOrString{IntVal: 8000},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	cfgs := map[string]*model.Config{}
+	ConvertIngressVirtualService(ingress, "mydomain", cfgs)
+	ConvertIngressVirtualService(ingress2, "mydomain", cfgs)
+
+	if len(cfgs) != 3 {
+		t.Error("VirtualServices, expected 3 got ", len(cfgs))
+	}
+
+	for n, cfg := range cfgs {
+		// Not clear if this is right - should probably be under input ns
+		if cfg.ConfigMeta.Namespace != "istio-system" {
```

Comment:
You cannot have an ingress controller per namespace. So it doesn’t matter which namespace we convert all the ingress resource into.

Reply:
Ack - yes, the secrets must be in same namespace with the gateway.

Reply:
There is also a constant for Istio-system FYI